tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Graph.java

```diff
@@ -374,6 +379,84 @@ public GraphOperationBuilder opBuilder(String type, String name) {
     return new GraphOperationBuilder(this, type, name);
   }
 
+  @Override
+  public void attachFunction(ConcreteFunction function) {
+    try (Reference ref = ref();
+        PointerScope scope = new PointerScope()) {
+      attachNativeFunction(ref.nativeHandle(), function.nativeHandle());
+      function.getDependencies().forEach(x -> attachNativeFunction(ref.nativeHandle(), x));
+    }
+  }
+
+  private void attachNativeFunction(TF_Graph graph, TF_Function fn) {
```

Comment:
maybe move the entire code of this mode in the unique place it is called (i.e. in `attachFunction`) so we are sure it is never being called outside a `PointerScope`?